generator/src/main/java/com/scottlogic/deg/generator/guice/RowSpecTreeSolverProvider.java

```diff
@@ -0,0 +1,37 @@
+package com.scottlogic.deg.generator.guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.scottlogic.deg.common.ValidationException;
+import com.scottlogic.deg.generator.generation.GenerationConfigSource;
+import com.scottlogic.deg.generator.walker.decisionbased.DecisionBasedSolver;
+import com.scottlogic.deg.generator.walker.rowspec.CartesianProductRowSpecTreeSolver;
+import com.scottlogic.deg.generator.walker.rowspec.RowSpecTreeSolver;
+
+public class RowSpecTreeSolverProvider implements Provider<RowSpecTreeSolver> {
+
+    private final GenerationConfigSource config;
+    private final CartesianProductRowSpecTreeSolver cartesianProductRowSpecTreeSolver;
+    private final DecisionBasedSolver decisionBasedSolver;
+
+    @Inject
+    public RowSpecTreeSolverProvider(GenerationConfigSource config, CartesianProductRowSpecTreeSolver cartesianProductRowSpecTreeSolver, DecisionBasedSolver decisionBasedSolver) {
+        this.config = config;
+        this.cartesianProductRowSpecTreeSolver = cartesianProductRowSpecTreeSolver;
+        this.decisionBasedSolver = decisionBasedSolver;
+    }
+
+    @Override
+    public RowSpecTreeSolver get() {
+        switch (config.getWalkerType()) {
+            case DECISION_BASED:
+                return decisionBasedSolver;
+            case CARTESIAN_PRODUCT:
+                return cartesianProductRowSpecTreeSolver;
+            case REDUCTIVE:
+                return decisionBasedSolver;
```

Comment:
I don't think the Decision Based Solver should be returned here.

Reply:
something not null needs to be returned, guice is a bit awkward when nothing is needed